### PR TITLE
Dispatching

### LINKED ISSUE #207: prevent requests performed offline from being lost
requests performed when client is disconnected should be cached and executed when getting connected again

### PR DESCRIPTION
**Pull request by llbartekll**
refactore Relayer
improves requests dispatching 
close #207

### CODE REVIEW

File: Sources/WalletConnectUtils/Queue.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        serialQueue.sync {
+            return elements.last
+        }
+    }
+    
+    public init(elements: [T] = []) {
+        self.elements = elements
+    }
+    
+    public func enqueue(_ value: T) {
```
Comment (MisterVants) [inline Sources/WalletConnectUtils/Queue.swift:24]: I'm more of a fan of `push()` & `pop()` semantics 😂

File: Tests/RelayerTests/DispatcherTests.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        XCTAssertEqual(webSocketSession.sendCallCount, 0)
+        sut.connect()
+        socketConnectionObserver.onConnect?()
+        XCTAssertEqual(webSocketSession.sendCallCount, 2)
+    }
+    
+    func testOnMessage() {
+        webSocketSession.onMessageReceived?("message")
+        sut.onMessage = { message in
+            XCTAssertNotNil(message)
```
Comment (MisterVants) [inline Tests/RelayerTests/DispatcherTests.swift:51]: The assertion is not being called on this test, you need to set the `onMessage` closure before calling `onMessageReceived`

**Review by MisterVants — CHANGES_REQUESTED** [review]
There's just one test case that needs to be fixed. Rest is all good.

File: Sources/WalletConnectUtils/Queue.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/WalletConnectUtils/Queue.swift:24]: `push` and `pop` sounds good but for stacks xd

Comment (MisterVants) [reply Sources/WalletConnectUtils/Queue.swift:24]: lol, I've confused queue with stacks

**Review by MisterVants — APPROVED** [review]

**Pull request merged**